power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/short_circuit_solver.hpp

```diff
@@ -235,7 +242,102 @@ class ShortCircuitSolver {
     typename SparseLUSolver<ComplexTensor<sym>, ComplexValue<sym>, ComplexValue<sym>>::BlockPermArray perm_;
 
-    void set_phase_index_(double& phase_1, double& phase_2, FaultPhase fault_phase) {
+    void calculate_result(YBus<sym> const& y_bus, ShortCircuitInput const& input, ShortCircuitMathOutput<sym>& output,
+                          IdxVector const& zero_fault_counter, FaultType const fault_type, int const& phase_1,
+                          int const& phase_2, double const& source_voltage_ref) {
+        // loop through all buses
+        for (Idx bus_number = 0; bus_number != n_bus_; ++bus_number) {
+            const ComplexValue<sym> x_tmp = output.u_bus[bus_number];  // save x to temp variable
+            const double zero_fault_counter_bus = static_cast<double>(zero_fault_counter[bus_number]);
+            for (Idx fault_number = (*fault_bus_indptr_)[bus_number];
+                 fault_number != (*fault_bus_indptr_)[bus_number + 1]; ++fault_number) {
+                const DoubleComplex y_fault = input.faults[fault_number].y_fault;
+                if (std::isinf(y_fault.real())) {
+                    assert(std::isinf(y_fault.imag()));
+                    if constexpr (sym) {  // three phase fault
+                        output.i_fault[fault_number] =
+                            -1.0 * x_tmp / zero_fault_counter_bus;  // injection is negative to fault
+                        output.u_bus[bus_number] = 0.0;
+                    }
+                    else if (fault_type == FaultType::single_phase_to_ground) {
+                        output.i_fault[fault_number](phase_1) = -1.0 * x_tmp[phase_1] / zero_fault_counter_bus;
+                        output.u_bus[bus_number](phase_1) = 0.0;
+                    }
+                    else if (fault_type == FaultType::two_phase) {
+                        output.i_fault[fault_number](phase_1) = -1.0 * x_tmp[phase_2] / zero_fault_counter_bus;
+                        output.i_fault[fault_number](phase_2) = -1.0 * output.i_fault[fault_number](phase_1);
+                        output.u_bus[bus_number](phase_2) = output.u_bus[bus_number](phase_1);
+                    }
+                    else {
+                        assert((fault_type == FaultType::two_phase_to_ground));
+                        output.i_fault[fault_number](phase_1) = -1.0 * x_tmp[phase_1] / zero_fault_counter_bus;
+                        output.i_fault[fault_number](phase_2) = -1.0 * x_tmp[phase_2] / zero_fault_counter_bus;
+                        output.u_bus[bus_number](phase_1) = 0.0;
+                        output.u_bus[bus_number](phase_2) = 0.0;
+                    }
+                }
+                else {
+                    assert(!std::isinf(y_fault.imag()));
+                    if (zero_fault_counter[bus_number] > 0) {
+                        // ignore fault objects with impedance, when there is a fault with infinite impedance on bus
```

Comment:
i'm not sure i like this name. please rename this to something like `has_infinite_impedance` or maybe even `should_skip_bus` and then add the reason in the implementation. usually if you need a comment that does not directly explain what the code does, you might as well make that comment the name of a new function.

also, shouldn't this be `infinite accceptance` instead of `infinite impedance`?

Reply:
renamed it to `infinite_admittance_fault_counter`